static/js/components/Editor_test.js

```diff
@@ -0,0 +1,351 @@
+    } call .focus on view if autoFocus = ${String(autoFocus)}`, () => {
+      renderEditor({ autoFocus })
+      assert.equal(focusStub.called, autoFocus)
+    })
+  })
+
+  it("should set the focus helper on the whole editor", () => {
+    const wrapper = renderEditor()
+    assert.equal(
+      wrapper.instance().setEditorFocus,
+      wrapper.find(".editor").props().onClick
+    )
+  })
+
+  it("should use the initialValue prop, if supplied", () => {
+    const initialValue = "# My Markdown Document\n\nJust a great doc"
+    const wrapper = renderEditor({ initialValue })
+    const output = defaultMarkdownSerializer.serialize(
+      wrapper.instance().view.state.doc
+    )
+    assert.equal(output, initialValue)
+  })
+
+  describe("dispatchTransaction", () => {
+    it("should bind our custom handler", () => {
+      const dispatchStub = sandbox.stub(Editor.prototype, "dispatchTransaction")
+      const wrapper = renderEditor()
+      executeTestTransaction(wrapper)
+      assert(dispatchStub.called)
+    })
+
+    it("should call the onChange handler", () => {
+      const wrapper = renderEditor()
+      executeTestTransaction(wrapper)
+      assert(onChange.calledWith(""))
+    })
+
+    it("should call onChange with serialized md", () => {
+      const mdStub = sandbox
+        .stub(pmMarkdown.defaultMarkdownSerializer, "serialize")
+        .returns("# MARKDOWN!")
+      executeTestTransaction(renderEditor())
+      assert(onChange.calledWith("# MARKDOWN!"))
+      assert(mdStub.called)
+    })
+  })
+
+  describe("menuClickHandlerCreator", () => {
+    it("menuClickHandlerCreator should return a function", () => {
+      assert.isFunction(
+        renderEditor()
+          .instance()
+          .menuClickHandlerCreator(sandbox.stub())
+      )
+    })
+
+    it("should call passed function with correct args, preventDefault, focus", () => {
+      const commandStub = sandbox.stub()
+      const preventDefault = sandbox.stub()
+      const wrapper = renderEditor()
+      const clickHandler = wrapper
+        .instance()
+        .menuClickHandlerCreator(commandStub)
+      clickHandler({ preventDefault })
+      assert(
+        commandStub.calledWith(
+          wrapper.instance().view.state,
+          wrapper.instance().view.dispatch,
+          wrapper.instance().view
+        )
+      )
+      assert(preventDefault.called)
+      assert(focusStub.called)
+    })
+  })
+
+  describe("menu bar", () => {
+    it("should have the things we tell it do", () => {
+      const wrapper = renderEditor()
+      assert.equal(
+        wrapper.find(".button-group").length,
+        menuBarManifest.length + 1
+      )
+      const allButtons = wrapper.find(".menu-button")
+      assert.equal(allButtons.length, R.flatten(menuBarManifest).length + 1)
+      assert.deepEqual(allButtons.map(button => button.text()), [
+        ...R.flatten(menuBarManifest).map(item => item.icon.props.children),
+        "link"
```

Comment:
Can you create variables for these lists so it's a little more readable & clearer about what you're comparing?